source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -157,6 +159,12 @@ bool PolicyMatcher::matches(const Network::Connection& connection,
          principals_.matches(connection, headers, metadata);
 }
 
+bool RequestedServerNameMatcher::matches(const Network::Connection& connection,
+                                         const Envoy::Http::HeaderMap&,
+                                         const envoy::api::v2::core::Metadata&) const {
+  return connection.requestedServerName().compare(server_name_) == 0;
```

Comment:
is this a string equals function? For SNI specifically, we need prefix matches at lest. (like *.example.com) that matches foo.example.com and so on.. Didn't RBAC have some exact/prefix/regex match construct for strings/